postnet/src/postnet.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Created by SONY on 2016/7/26.
+ */
+let _ = require('lodash');
+function _getBarcodes(){
+    return ['||:::',':::||','::|:|','::||:',':|::|',':|:|:',':||::','|:::|','|::|:','|:|::'];
+}
+function checkPostCode(postcode){
+    let flag = false;
+    if(postcode.length ===5 || postcode.length === 9 || (postcode.length === 10 && postcode.charAt(5) === '-')){
+        postcodeno = postcode.replace('-','').split('');
+        let temp = postcodeno.map((element)=>{
+            let reg =/\d/;
+            return reg.test(element)
+        });
+        if(!temp.includes(false)){
+            flag = true;
+        }
+    }
+    if(flag){
+        return postcode;
+    }else{
+        return flag;
+    }
+}
+function formatPostCode(postcode){
+    if(postcode !== false){
+        if(postcode.includes('-')){
+            let temp = postcode.split('-');
+            postcode = temp[0]+temp[1];
+        }
+    }
+    return postcode;
+}
+function matchBarcodes(postcode,barcodes){
+    let postcodes = postcode.split('');
+    let postcodesInt = postcodes.map((postcode)=>{
+        return parseInt(postcode);
+    });
+    let sum = _.sum(postcodesInt);
+    if(sum % 10 === 0){
+        postcodes.push('0');
+    }else{
+        postcodes.push(parseFloat( ));
+    }
+    return postcodes.map((postcode)=>{
+        return  barcodes[parseInt(postcode)];
+    });
+}
+function getBarcodeString(barcodes){
+    let barcodeString = '|';
+    let codesString = barcodes.join('');
+    barcodeString += codesString;
+    barcodeString += '|';
+    return barcodeString;
+}
+//条码转邮编
+function getcodesObjects(){
+    return [
+        {no:'0',code:'||:::'},{no:'1',code:':::||'},{no:'2',code:'::|:|'},{no:'3',code:'::||:'},{no:'4',code:':|::|'},
+        {no:'5',code:':|:|:'},{no:'6',code:':||::'},{no:'7',code:'|:::|'},{no:'8',code:'|::|:'},{no:'9',code:'|:|::'}
+    ];
+}
+function checkBarcode(barcode){
+    if(barcode.length === 32 || barcode.length === 52  ){
+        if( barcode.charAt(0)==='|' && barcode.charAt(barcode.length-1)==='|'){
+            let temp = barcode.split('').map((element)=>{
+                let reg =/\d/;
+                return reg.test(element); //匹配有其他字符,数字，特殊字符等
+            });
+            if(!temp.includes(true)){
+                return barcode;
```

Comment:
是的。我现在又重新更新了，多加了判定条件，你可以看新的，是在这基础上添加的

在 2016-07-30 12:02:27，"ZhuoYueYue" <notifications@github.com> 写道：


In postnet/src/postnet.js:

> +//条码转邮编
> +function getcodesObjects(){
> +    return [
> +        {no:'0',code:'||:::'},{no:'1',code:':::||'},{no:'2',code:'::|:|'},{no:'3',code:'::||:'},{no:'4',code:':|::|'},
> +        {no:'5',code:':|:|:'},{no:'6',code:':||::'},{no:'7',code:'|:::|'},{no:'8',code:'|::|:'},{no:'9',code:'|:|::'}
> +    ];
> +}
> +function checkBarcode(barcode){
> +    if(barcode.length === 32 || barcode.length === 52  ){
> +        if( barcode.charAt(0)==='|' && barcode.charAt(barcode.length-1)==='|'){
> +            let temp = barcode.split('').map((element)=>{
> +                let reg =/\d/;
> +                return reg.test(element); //匹配有其他字符,数字，特殊字符等
> +            });
> +            if(!temp.includes(true)){
> +                return barcode;


如果它temp里有其他字符，元素里就有false是吗

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.